my-code/src/containers/Detail.js

```diff
@@ -0,0 +1,214 @@
+
+import { Text100, Text200, Text400 } from "../components/Typography/styled"
+
+type IdType = string
+
+type PropsType = {
+  match: {
+    params: {
+      id: string
+    }
+  }
+}
+
+type StateType = {
+  info: {
+    Title?: string,
+    Runtime?: string,
+    Genre?: string,
+    Plot?: string,
+    Rated?: string,
+    Poster?: string,
+    Director?: string,
+    Actors?: string,
+    Year?: string,
+    Ratings?: {
+      Source: string,
+      Value: string
+    }
+  }
+}
+
+type ActorType = {
+  actor: string
+}
+type GenreType = {
+  actor: string
+}
+
+type DirectorType = {
+  actor: string
+}
+
+type RatingsType = {
+  Source: string,
+  Value: string
+}
+
+class Detail extends React.Component<PropsType, StateType> {
+  constructor() {
+    super()
+    this.state = {
+      info: {},
+      favourite: false,
+      active: false
+    }
+  }
+
+  componentDidMount() {
+    const {
+      match: {
+        params: { id }
+      }
+    } = this.props
+
+    this.loadMovies(id)
+  }
+
+  loadMovies = async (id: IdType) => {
+    const url = `${omdbApi.BASE_URL}${omdbApi.API_KEY}&i=${id}`
+    const res = await fetch(url)
+    const data = await res.json()
+
+    this.setState({ info: data || {} })
+  }
+
+  saveToFavourites = () => {
+    const { favourite } = this.state
+
+    this.setState(
+      prevState => ({
+        favourite: !prevState.favourite,
+        active: !prevState.active
+      }),
+      () => localStorage.setItem("favourite", !favourite)
+    )
+  }
+
+  render() {
+    const { info } = this.state
+    const { active } = this.state
+
+    console.log(info)
+
+    return (
+      <Wrapper width={1180}>
+        <FlexLeft width={1 / 2}>
+          <Link to="/">
+            <Arrow />
+          </Link>
+
+          <Wrapper mt={6} mb={4}>
+            <Text200 grey>
+              {info.Runtime ? info.Runtime : "Unknown runtime"} ·&nbsp;
+            </Text200>
+            <Text200 grey>
+              {info.Year ? info.Year : "Unknown year"} ·&nbsp;
+            </Text200>
+            <Text200 grey>{info.Rated ? info.Rated : "No rating"}</Text200>
+          </Wrapper>
+
+          <Text400 mb={5}>{info.Title}</Text400>
+          {info.Ratings ? (
+            <Wrapper mb={5}>
+              {info.Ratings.map((ratings: RatingsType) => (
+                <div>
+                  <div>
+                    {ratings.Source === "Internet Movie Database" && (
+                      <IMDBLabel value={ratings.Value} />
+                    )}
+                  </div>
+                  <div>
+                    {ratings.Source === "Rotten Tomatoes" && (
+                      <RottenLabel value={ratings.Value} />
+                    )}
+                  </div>
+                </div>
+              ))}
+              <Button
+                text="Add to favourites"
+                onClick={this.saveToFavourites}
+                active={active}
+              />
+            </Wrapper>
+          ) : null}
```

Comment:
If it's a conditional render no need to use ternaries and return null. you could just `info.Ratings &&`